sdk/tdferrors.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	errFileTooLarge     = errors.New("tdf: can't create tdf larger than 64gb")
+	errWriteFailed      = errors.New("tdf: io.writer fail to write all bytes")
+	errInvalidKasInfo   = errors.New("tdf: kas information is missing")
+	errKasPubKeyMissing = errors.New("tdf: kas public key is missing")
+	errRewrapForbidden  = errors.New("tdf: rewrap request 403")
+	// Exposed tamper detection errors, Catch all possible tamper errors with errors.Is(ErrTampered)
+	ErrTampered                = errors.New("tamper detected")
+	ErrRootSigValidation       = fmt.Errorf("[%w] tdf: failed integrity check on root signature", ErrTampered)
+	ErrSegSizeMismatch         = fmt.Errorf("[%w] tdf: mismatch encrypted segment size in manifest", ErrTampered)
+	ErrTDFReaderFailed         = fmt.Errorf("[%w] tdf: fail to read bytes from TDFReader", ErrTampered)
```

Comment:
I don't think a read failure would indicate tampering.

Reply:
ya i think this one was because of the way it was used when the segment size didnt match the buf len (see above comment) but if we switch those to ErrSegSizeMismatch as suggested then this can be removed from the tamper errors